rpc/namespaces/ethereum/eth/filters/filter_system.go

```diff
@@ -153,14 +146,30 @@ func (es *EventSystem) subscribe(sub *Subscription) (*Subscription, pubsub.Unsub
 		return nil, nil, err
 	}
 
-	// wrap events in a go routine to prevent blocking
-	es.install <- sub
-	<-sub.installed
+	// add subscription to the index
+	es.index[sub.typ][sub.id] = sub
```

Comment:
Shouldn't we add subscription to index AFTER we `AddTopic`? This way, if `AddTopic` returns an error, we don't need to do `delete(es.index[sub.typ], sub.id)`

Reply:
sure thing

Reply:
9986459c18656c8058b69c2619225adb18c462fb